index.js

```diff
@@ -46,15 +46,40 @@ class CoverageBase {
     /**
      * Get shell command to upload coverage to server
      * @method getUploadCoverageCmd
-     * @return {Promise}     Shell commands to upload coverage
+     * @param   {Object}  config
+     * @param   {String}  config.build   build configuration
+     * @return {Promise}  Shell commands to upload coverage
      */
-    getUploadCoverageCmd() {
+    getUploadCoverageCmd(config) {
+        if (this.isCoverageEnabled(this.config, config.build) === 'false') {
+            const skipMessage = 'Coverage feature is skipped. ' +
+                'Set COVERAGE_PLUGIN_ENABLED environment true, if you want to get coverages.';
+
+            return Promise.resolve(`echo ${skipMessage}`);
+        }
+
         return this._getUploadCoverageCmd();
     }
 
     _getUploadCoverageCmd() {
         return Promise.reject('Not implemented');
     }
+
+    /**
+     * Verify to run coverage plugin or not
+     * @method isCoverageEnabled
+     * @param   {Object}  clusterConfig  coverage plugin setting at cluster level.
+     * @param   {Object}  buildConfig.environment.COVERAGE_PLUGIN_ENABLED coverage plugin setting in each builds.
+     * @return {Boolean}
+     */
+    isCoverageEnabled(clusterConfig, buildConfig) {
```

Comment:
How about these code (it's just dummy codes, need more variables check)?  I think it's simpler.
```
// if user has the configuration, use it
if (buildConfig.environment.COVERAGE_PLUGIN_ENABLED) {
    return buildConfig.environment.COVERAGE_PLUGIN_ENABLED
}
// if not, use cluster wide default
return clusterConfig.enabled;
```

Reply:
Thanks, I fixed like above.
There is a case that `buildConfig.environment` property itself dose not exist. So, I checked that case.